lib/ecto/adapters/myxql.ex

```diff
@@ -238,7 +238,7 @@ defmodule Ecto.Adapters.MyXQL do
         {:ok, last_insert_id(key, last_insert_id)}
 
       {:error, err} ->
-        case @conn.to_constraints(err) do
+        case @conn.to_constraints(err, [source: extract_tablename(sql)]) do
```

Comment:
have you tried using `[source: source]` here?

Reply:
Oh yes. Very nice. I can even drop my function for messing with the sql to get the tablenames. Thanks. 

I have pushed an updated commit. 

I will also have a look if i can fix the CI script, mysql is now working but postgres has still an issue installing the client.

Reply:
The CI script is fixed, the tests for postgres are also running now.